core/object.rbs

```diff
@@ -205,8 +205,8 @@ class Object < BasicObject
   #     enum.first(4) # => [1, 1, 1, 2]
   #     enum.size # => 42
   #
-  def enum_for: (Symbol method, *untyped args) ?{ (*untyped args) -> Integer } -> Enumerator[untyped, untyped]
-              | (*untyped args) ?{ (*untyped args) -> Integer } -> Enumerator[untyped, untyped]
+  def enum_for: (Symbol method, *untyped args) ?{ (*untyped args) -> Integer } -> Enumerator[untyped, self]
```

Comment:
I don't think we can assume `Enumerator[untyped, self]` here, because the `enum_for` can be used for any methods.

Reply:
Good point, I'll revert this.

Reply:
I was just thinking on this, I realised the main reason I added this was because `to_enum` is aliased here, so the use-case where it's useful is for when someone calls `some_obj.to_enum` with no arguments - this defaults to `each` and returning `self`.

Would the following signature be okay by you? Here `self` is the return type when no method symbol is provided, and in the other situation we use `untyped`.

```
  def enum_for: (*untyped args) ?{ (*untyped args) -> Integer } -> Enumerator[untyped, self]
              | (Symbol method, *untyped args) ?{ (*untyped args) -> Integer } -> Enumerator[untyped, untyped]
```

Reply:
I agree that most (all?) `#each` implementation returns `self` and assuming `self` is reasonable.

Minor fix: if we omit the `method` argument, we cannot pass any other argument?

```
def enum_for: () ?{ () -> Integer } -> Enumerator[untyped, self]
            | ~~~
```

Reply:
Seems you are correct, done.